packages/eslint-mdx/src/types.ts

```diff
@@ -1,6 +1,14 @@
 import type { JSXElement, JSXFragment } from '@babel/types'
 import type { AST, Linter } from 'eslint'
-import type { Node, Parent, Point } from 'unist'
+import type { Node as _Node, Parent as _Parent, Point } from 'unist'
```

Comment:
It might be better to use `@types/mdast` here, and extend it with the nodes added by `@mdx-js/mdx@1`

Reply:
We only care about `mdx` types

```ts
export const ES_NODE_TYPES: readonly string[] = ['export', 'import', 'jsx']
```

Reply:
Then you could type those here?

Reply:
Also: I do think you probably should use `@types/mdast`, with the three new MDX types, here 🤔

Reply:
For what? `Node & { value: string }` is all needed. I don't want to add unnecessary types.

Reply:
You can import just the things you need from `@types/mdast` and drop `@types/unist`?

Reply:
`@types/mdast` depends on `@types/unist`.

```ts
export interface Parent extends UnistParent {
   children: Content[];
}
```

The `Parent` is not what I need, I still have to extend it and override `children` property. It's just the same, and few deps.

Reply:
@wooorm Any further to discuss? @ChristianMurphy proposed to improve the inner types in the future at https://github.com/mdx-js/eslint-mdx/pull/321#discussion_r667132411, and it should not be the blocker IMO.

After this PR, [`validate-links`](https://github.com/remarkjs/remark-validate-links) can be enabled in [mdx](https://github.com/mdx-js/mdx/blob/main/.remarkrc.js#L7) because the performance is much better now. And I just found some missing links of `license` locally.

https://github.com/mdx-js/mdx/blob/main/packages/test-util/readme.md for example.

Reply:
Nothing to discuss, except that I remain of the opinion that you should use mdast and extend it.